common/lib/xmodule/xmodule/capa_module.py

```diff
@@ -4,50 +4,267 @@
 import re
 import sys
 
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
 from lxml import etree
 from pkg_resources import resource_string
+from web_fragments.fragment import Fragment
+from xblock.core import XBlock
+from xblock.fields import Boolean, Dict, Float, Integer, Scope, String, XMLString
 
 from capa import responsetypes
+from xmodule.editing_module import EditingMixin
 from xmodule.exceptions import NotFoundError, ProcessingError
-from xmodule.raw_module import RawDescriptor
+from xmodule.graders import ShowCorrectness
+from xmodule.raw_module import RawMixin
 from xmodule.contentstore.django import contentstore
 from xmodule.util.misc import escape_html_characters
 from xmodule.util.sandboxing import get_python_lib_zip
-from xmodule.x_module import DEPRECATION_VSCOMPAT_EVENT, XModule, module_attr
+from xmodule.util.xmodule_django import add_webpack_to_fragment
+from xmodule.x_module import (
+    HTMLSnippet, ResourceTemplates, shim_xmodule_js,
+    XModuleMixin, XModuleToXBlockMixin, XModuleDescriptorToXBlockMixin,
+)
+from xmodule.xml_module import XmlParserMixin
 
-from .capa_base import CapaFields, CapaMixin, ComplexEncoder
+from .capa_base import _, CapaMixin, ComplexEncoder, RANDOMIZATION, Randomization, SHOWANSWER
+from .fields import Date, Timedelta, ScoreField
 
 log = logging.getLogger("edx.courseware")
 
 
-class CapaModule(CapaMixin, XModule):
-    """
-    An XModule implementing LonCapa format problems, implemented by way of
-    capa.capa_problem.LoncapaProblem
+try:
+    FEATURES = getattr(settings, 'FEATURES', {})
+except ImproperlyConfigured:
+    FEATURES = {}
+
 
-    CapaModule.__init__ takes the same arguments as xmodule.x_module:XModule.__init__
+@XBlock.wants('user')  # pylint: disable=abstract-method
+@XBlock.needs('i18n')
+class ProblemBlock(
+        CapaMixin, RawMixin, XmlParserMixin, EditingMixin,
+        XModuleDescriptorToXBlockMixin, XModuleToXBlockMixin, HTMLSnippet, ResourceTemplates, XModuleMixin):
     """
+    Define the possible fields for a Capa problem
+    """
+    display_name = String(
```

Comment:
The fields were moved in from the `CapaFields` class. The only changes are a couple of tweaks for quality issues.